modules/button/react/lib/Button.tsx

```diff
@@ -36,7 +42,9 @@ export interface ButtonProps
   icon?: CanvasSystemIcon;
 }
 
-const Button = ({
+function Button(props: ButtonProps): React.ReactElement;
```

Comment:
This is the old/standard way to do function overloading. I explored how to do it with arrow functions as well:
```tsx
type ButtonOverload = {
  (props: ButtonProps): React.ReactElement;
  (props: {as: 'a'} & AnchorButtonProps<ButtonProps>): React.ReactElement;
  Variant: typeof ButtonVariant;
  Size: typeof ButtonSize;
};
const Button: ButtonOverload = ({
  theme = useTheme(),
  variant = ButtonVariant.Secondary,
  size = 'medium',
  buttonRef,
  dataLabel,
  icon,
  children,
  ...elemProps
}: ButtonProps) => (
  <ButtonContainer
    colors={getButtonColors(variant, theme)}
    size={size}
    ref={buttonRef}
    {...elemProps}
  >
    {icon && size !== 'small' && <ButtonLabelIcon size={size} icon={icon} />}
    <ButtonLabel>{children}</ButtonLabel>
    {dataLabel && size !== 'small' && <ButtonLabelData>{dataLabel}</ButtonLabelData>}
  </ButtonContainer>
);
```

I think I lean towards the arrow function for consistencies sake (even though you need to define the static variables again. Anyone have a preference??

Reply:
I'm good with an arrow function. It seems like our docgen understands the props?